### ISSUE TITLE
Plan for WDL 1.2 support?

### ISSUE THREAD

**Issue by yunhailuo** [chanzuckerberg/miniwdl#703]
First, thank you for the great engine and tools here! We have been benefiting from your hard work here for a while 🙏 

Second, I've tried searching code, PR, issues, projects and didn't fine much info so could have missed; apologize if so.

It seems WDL 1.2 has been nailed and released for a month or so. I hope it's not too early to ask but is there a plan/roadmap for miniwdl to start supporting WDL 1.2? Technically, miniwdl support `development` which is hopefully closer to 1.2 than 1.0. But I know the devil is in the detail so just want to kindly ask for plan rather than results 😄 (Just in case this helps in any way, happy to contribute though I doubt how much I can be useful 😅 ).

**Comment by mlin** [comment]
Yes, here is the plan to make a plan. Right now I've been focusing on some "meta" technical debt, especially #668. That's going to be equally important to keeping the project healthy in the long term. Then I will make a thorough pass through the new spec and file tickets, then start burning them down, where any contributions would certainly be welcome. Best guess, I'd hope to ship it all towards the end of Q3.

As you note, I think the most complex of the new features are already done (`Directory` and multi-line strings), but there's certainly a tail of other stuff to do.

**Comment by yunhailuo** [comment]
Thank you and truly appreciate the details and timeline. This is exciting! Close here as this is just a question. Still happy to help if feasible.

**Comment by patmagee** [comment]
Hey @mlin i just wanted to come back to this and see what support you need from the community to get 1.2 over the line.

I think there are a number of us that would be happy to contribute some time if we know the checklist

**Comment by claymcleod** [comment]
I am also very curious for your plans on this, @mlin.

**Comment by vsmalladi** [comment]
Maybe worth running miniwdl through compliance for wdl 1.2 spec and make subissues?

**Comment by yunhailuo** [comment]
@mlin I just noticed https://github.com/chanzuckerberg/miniwdl/milestone/6 today. I wonder what's your plan on that? Like do you plan to call for PR contribution?

**Comment by mlin** [comment]
@yunhailuo now that we're in the singularity, I think Claude/Codex/etc will advance to the stage where it can finish it before I would by typing in the code (70% serious)

**Comment by yunhailuo** [comment]
> [@yunhailuo](https://github.com/yunhailuo) now that we're in the singularity, I think Claude/Codex/etc will advance to the stage where it can finish it before I would by typing in the code (70% serious)

Oh. Nice, you already have AI agents working on them 😄 I'll stay put and look forward to it.

**Comment by mlin** [comment]
[miniwdl v1.15.0rc1](https://github.com/chanzuckerberg/miniwdl/releases) completes support for WDL 1.2. We plan to spend at least a few weeks in "release candidate" cycle, since it's the largest change to the codebase in some years. We still have the opportunity to rework things based on any feedback in the meantime.

**Comment by vsmalladi** [comment]
@mlin  amazing let us know how we can help test.

**Comment by claymcleod** [comment]
Thanks for all your hard work on this Mike!

**Comment by mlin** [comment]
@vsmalladi I'd be really interested in an independent assessment of this build on the spec conformance tests. We have a custom spec test runner with a [small xfail list](https://github.com/chanzuckerberg/miniwdl/blob/d57d5934a0ea6d5e22fefe23776343401ffa7049/tests/spec_tests/config.yaml#L23-L39) but another set of eyes/hands/AIs on it could well uncover something we missed.

**Comment by claymcleod** [comment]
I'm not sure if you've seen this yet, @mlin, but we recently released specification testing in the OpenWDL CI using [`spectool`](https://github.com/openwdl/spectool). There's a line in the README for using it with miniwdl. I'd highly recommend leaning on that, as we'll keep it in sync with the spec long term.

**Comment by vsmalladi** [comment]
@mlin  ran (spectool)[https://github.com/openwdl/spectool] against the `v1.15.0rc1`.

[miniwdl-spectool-logs.txt](https://github.com/user-attachments/files/29974981/miniwdl-spectool-logs.txt)

## MiniWDL Spectool Run Summary (WDL 1.2)

### Result Overview
- Total tests: **170**
- Passed: **160**
- Failed: **10**
- Skipped: **1**

### Comparison Against MiniWDL Known Expectations

#### Matches Known `xfail`
Theses match MiniWDL’s known broken/deviation list:
- one_mount_point_task.wdl
- placeholder_none.wdl
- test_object.wdl
- read_object_task.wdl
- read_objects_task.wdl
- read_tsv_task.wdl
- write_object_task.wdl
- write_objects_task.wdl

Count: **8/10 failures**

#### Known `xfail` That Passed
- file_sizes_task.wdl (**passed**)

### Skip/Expectation Mismatches

#### Expected `skip` and did skip
- test_gpu_task.wdl

#### Expected `skip` but behavior differed
- multi_nested_inputs.wdl: ran and failed with “test marked with `fail: true` but succeeded”
- write_json_fail.wdl: ran and failed with “test marked with `fail: true` but succeeded”

Count: **2 mismatches**

### Conclusion
Most failures are consistent with MiniWDL’s known limitations.  
The main delta is skip-policy mismatch for:
- multi_nested_inputs.wdl
- write_json_fail.wdl
